lib/Web/Main.hs

```diff
@@ -4,18 +4,75 @@
 
 module Web.Main
   ( runServer
+  , RunServerException (..)
   ) where
 
 import Backend.Commands (runCommand)
 import Config
+import Control.Exception (Exception, throw)
+import Data.Bifunctor (Bifunctor(first))
+import Data.Either (partitionEithers)
 import Data.Proxy
 import Network.Wai.Handler.Warp (run)
+import Options.Applicative
 import Servant.Server (serve)
+import System.Environment (lookupEnv)
+import Text.Interpolation.Nyan
+import Text.Read (readEither)
 import Web.API (API)
 import Web.Server
 
+data ServerOptions = ServerOptions { serverPort :: Maybe Int }
+
+data RunServerException
+  = RunServerIncorrectPort Int
+  | RunServerNoPortSpecified
+  | RunServerEnvVarParseFail String
+  deriving stock (Eq)
+
+instance Show RunServerException where
+  show (RunServerIncorrectPort port) = "Port number should be greater then 0. Your port number is: " ++ show port
+  show RunServerNoPortSpecified = "No port where not specified. Can be set by envirment varieble \"COFFER_SERVER_PORT\" or by \"--port=$port_num\" cmd argument."
```

Comment:
```suggestion
  show RunServerNoPortSpecified = "No port was specified. It can be set using the environment variable \"COFFER_SERVER_PORT\" or the \"--port=$port_num\" command-line option."
```

Reply:
done